aptos-move/move-examples/token_objects/inscriptions-as-state/sources/inscriptions.move

```diff
@@ -0,0 +1,157 @@
+module inscriptions::inscriptions {
+    use std::error;
+
+    use aptos_framework::event;
+    use aptos_framework::object::{Self, ConstructorRef, Object};
+
+    use aptos_token_objects::token;
+
+    /// The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The inscription does not exist
+    const EINSCRIPTION_DOES_NOT_EXIST: u64 = 2;
+
+    #[event]
+    struct InscriptionMintEvent has drop, store {
+        inscription_id: u64,
+        object: Object<InscriptionData>,
+    }
+
+    struct InscriptionData has key {
+        inscription_id: u64,
+        data: vector<u8>,
+    }
+
+    #[resource_group(scope = module_)]
+    struct InscriptionStateGroup { }
+
+    #[resource_group_member(group = inscriptions::inscriptions::InscriptionStateGroup)]
+    struct InscriptionState has key {
+        next_inscription_id: u64,
+    }
+
+    public fun add_inscription(
+        constructor_ref: &ConstructorRef,
+        data: vector<u8>,
+    ): u64 acquires InscriptionState {
+        assert!(
+            object::object_exists<token::Token>(object::address_from_constructor_ref(constructor_ref)),
+            error::not_found(ETOKEN_DOES_NOT_EXIST),
+        );
+        let object_signer = object::generate_signer(constructor_ref);
+
+        let inscription_id = get_next_inscription_id();
+        let inscription_data = InscriptionData { inscription_id, data };
+        move_to(&object_signer, inscription_data);
+
+        let inscription_mint_event = InscriptionMintEvent {
+            inscription_id,
+            object: object::object_from_constructor_ref(constructor_ref),
+        };
+        event::emit(inscription_mint_event);
```

Comment:
Should we have some reference to the &object_signer so that we can use this event to find out where the inscription data is